app/uk/gov/hmrc/apisubscriptionfields/model/Responses.scala

```diff
@@ -22,14 +22,14 @@ import play.api.libs.json.{JsObject, Json}
 case class SubscriptionFieldsIdResponse(fieldsId: SubscriptionFieldsId)
 case class SubscriptionFieldsResponse(fieldsId: SubscriptionFieldsId, fields: Fields)
 
+case class FieldsDefinitionResponse(fields: Seq[FieldDefinition])
 
 object ErrorCode extends Enumeration {
   type ErrorCode = Value
 
   val INVALID_REQUEST_PAYLOAD = Value("INVALID_REQUEST_PAYLOAD")
   val UNKNOWN_ERROR = Value("UNKNOWN_ERROR")
   val SUBSCRIPTION_FIELDS_ID_NOT_FOUND = Value("SUBSCRIPTION_FIELDS_ID_NOT_FOUND")
```

Comment:
I noticed you used this code for these two cases:
- subscription fields not found by id
- subscription fields not found by fieldsId

We need to use a different code or message.
Although in both cases we have a 404 because the subscription fields are not found in the mongo collection, in one case the id is not found, in the other case it is the fieldsId that is not found.

Reply:
There are now two separate error responses. 
- For a missing id the error message is `Id not found`
- For a missing fieldsId the error message is `FieldsId not found`